### ISSUE TITLE
Wash your hands

### ISSUE THREAD

**Issue by I-am-Erk** [CleverRaven/Cataclysm-DDA#62361]
### Is your feature request related to a problem? Please describe.

There are very few ways to limit common illnesses like colds in this game.

### Solution you would like.

In the midst of my big system advice and stable pushing, here's a fun aside.

We should have a "santizer dispenser" item, which can be crafted from almost any small bottle and loaded with hand sanitizer, alcohol, and other appropriate disinfectants. If you have it in your inventory, the charge gradually depletes (1 charge per hour or so) and your chance of getting sick with common illnesses is reduced by a small but significant amount, 50% or so. That is probably overstating the effectiveness of sanitizer, but since it's just to prevent a minor problem I don't see why not make it good.

For extra fun, we could add "compulsive handwashing" as a trait, where you go through sanitizer much more quickly and if you don't have it, suffer an increasing morale penalty.

### Describe alternatives you have considered.

There are other things we could add, but this would be fun and pretty easy.

Obviously we could add an active "wash your hands" job, but the micromanagement would be obnoxious.

### Additional context

Other things we could have:
- a hidden effect after interacting with an NPC or being in melee with a zombie/feral that increases illness chance for 24 hours
- that's really it actually. By this point I think we all know how hard it is to prevent droplet transmission of common viruses, eh?

We could also consider slightly reducing infection risk if you have a sani dispenser in your inventory when you do healthcare.

**Comment by I-am-Erk** [comment]
Side note, the general mechanic of "item in inventory slowly loses charges and gives a small benefit" could be used for a bunch of other minor things like this and would be an interesting way to implement some of our idle actions that are currently active. Especially if we expand "inventory" to include "nearby zones", so you can keep your hand sani on the desk.

**Comment by kevingranade** [comment]
Tangently related it doesn't look like my desire to move the cold and flu infection vector to NPCs and zombies has made it into an issue so just mentioning it.

**Comment by I-am-Erk** [comment]
IMO it's probably just as easy to make it a background thing, and treat "was close to someone who could have a virus" as an increased risk factor. That is probably a lot easier than tracking the actual infection. Maybe if someone has actually got a cold (assuming we let NPCs have colds) we could make them much riskier to be around.

**Comment by WarBarista** [comment]
Also, slowly drinking water while it is located in your inventory seems really good, especially during summer. I guess the "sanitizer draining system" can be applied to this. (Sorry for bad english)

**Comment by PatrikLundell** [comment]
Slowly drinking your carried water passively when it's hot is really bad when the only reason you notice that you're out of water 5 hours away from the base is that you're getting thirsty when passive drinking should keep that away. You need to keep track of your water consumption and head back in time.

Regarding the original suggestion, there's a risk of this being the start of a sliding slope of "huge pile of stuff you have to carry with you at all time or get penalized for not hauling it around with you". Not to mention the chore of topping up all the containers every day to make sure you don't run out. You probably can't leave your extra backpack of mandatory stuff in your vehicle either, because the checks probably won't be advanced enough to defer consumption until you're back in proximity of the stuff.

**Comment by I-am-Erk** [comment]
Similar to sanitizer as proposed, I'd suggest an auto drain water system would have to be actively declared by putting it into a particular type of container. Then you'd have the old function if desired, or you could declare a canteen an "auto canteen" and drink from it passively. It doesn't seem like a terrible idea.

Most of your concerns can be resolved with "just don't code it that way", Patrick... For example, keeping it in your car should be pretty valid, so we should ensure that works. In general the hand wash checks should run more often than every hour, so that if you leave you passive supplies somewhere and the buff runs out you'll use them when you get back. That's pretty easy to write.

**Comment by NetSysFire** [comment]
There could also be a mysophobia/germophobia trait, which requires you to wash your hands frequently or you will suffer heavy morale penalties and panic attacks.

And washing your hands so frequently will also cause skin irritation (could work with the new wound system) and maybe even morale penalties when its cold. Or the reverse when its hot.

**Comment by PatrikLundell** [comment]
As a minimum, you'd need to be notified when your supplies (of whatever) runs out. It would definitely be useful to be notified that you need to top them up when returning to base (whatever that is, which makes it tricky to code) and before heading out (again, what's the criterion for that?), optionally with an auto topping up system, again with notifications, but in this case not when things have run out, but when they're running low (replacing ethanol from scratch takes time, with fermentation and processing, for instance).

It's easy to say "just don't code it like that", but in practice, when it has already been coded like that (because of an oversight, typically, but also when a quick hack just gets left behind because the most basic functionality is there) developer reactions tend to be "it's working sufficiently well, and there are more pressing/interesting things to deal with".

There certainly is a large pile of stuff you want to have reasonably close at hand. I tend to dedicate a tile of vehicle storage for such things. It's common (at least for me) to forget to replace what's been used up (such as e.g. topping up used ammo mags).

**Comment by anoobindisguise** [comment]
This is especially good since gas masks don't provide absolute protection against sickness now. Sanitation is important, and while ideally there'd be a way to wash your hands with soap and water too it could just use soapy water charges instead of sanitizer.

**Comment by Zireael07** [comment]
Hm. I seem to remember similar ideas being proposed in the past and closed because "they add tedium/inventory management for little gain" (paraphrasing). How does this avoid this problem?

**Comment by I-am-Erk** [comment]
Washing hands as a mechanic would add tedium; I think having passive charged items that confer a small bonus would be a good alternative.

I'm not opposed to having item settings like "alert when empty". I'd suggest refills should be done from a refill zone as part of a "sort loot" action, so that you can avoid refilling it with your nice whisky automatically.

**Comment by estebandellasilva** [comment]
Generally not a bad idea ... but i think most people fairly quickly wear some gloves, which would also migitate some if not a lot of those problems 
+ Isnt it considered kinda bad if you keep  things to clean - especially for children who have to develop their immune system and for humans if you use desinfectant to much you can remove healthy bacteria from you skin and damage it
If you want to model one thing you would kinda need to also model the other thing

**Comment by Zireael07** [comment]
True, but I don't see "too clean" as an issue in post-apoc conditions like Cata.

**Comment by I-am-Erk** [comment]
Sanitation theory of allergy is much less well demonstrated than germ theory, and we're not going to model anything that develops gradually over years starting with childhood exposure.

Wearing gloves doesn't prevent exposure unless you're regularly disposing of them, which would need to be modeled the same way. Sanitizer is much more portable and generally more practical to represent

**Comment by epsimpson** [comment]
Horrible idea.  Needless fiddly bits that add zero to the game.  Illness and infection are great, so long as there's a cure that we can learn to take early, making it more effective, but constant prophylaxis is just dumb and useless, in game even more than real life.  Don't do it.  The morale penalty for wearing filthy is enough.

**Comment by I-am-Erk** [comment]
I still thikn this is a cute idea but after this much time it is safe to say nobody is biting
